core/amber/src/main/python/core/python_worker.py

```diff
@@ -30,13 +52,53 @@ def run(self) -> None:
             target=self._network_sender.run, name="network_sender"
         )
         main_loop_thread = Thread(target=self._main_loop.run, name="main_loop_thread")
+        self.original_parent_pid = os.getppid()
 
+        loop = asyncio.new_event_loop()
+        heartbeat_thread = Thread(
+            target=self.start_asyncio_loop, name="heartbeat_thread", args=(loop, 5.0)
+        )
+
+        heartbeat_thread.start()
         network_sender_thread.start()
```

Comment:
The order looks a bit strange. as heart beat would rely on the network sender to send a request, thus network sender should be started first?